src/allmydata/test/common_util.py

```diff
@@ -314,6 +314,11 @@ def getRemoteTubID(self):
     def getPeer(self):
         return "<fake>"
 
+    # For use by tests:
+    def disconnected(self):
+        for (f, args, kwargs) in list(self.disconnectors.values()):
+            f(*args, **kwargs)
+
```

Comment:
Maybe it's worth trying to catch double calls here?  I think that in real usage a canary can only be disconnected once.  If the tests somehow mis-use `disconnected` it might be nice to fail early.